ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,252 @@
+from collections import defaultdict
+from typing import Callable, cast, Dict, List, Optional, Tuple, Union
+
+import torch
+from torch.nn import functional as F
+
+import ignite.distributed as idist
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+class MeanAveragePrecision(Metric):
+    def __init__(
+        self,
+        iou_thresholds: Optional[Union[List[float], torch.Tensor]] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+        r"""Calculate the mean average precision of overall categories.
+
+        Args:
+            iou_thresholds: list of IoU thresholds to be considered for computing Mean Average Precision.
+                Values should be between 0 and 1. List is sorted internally. Default is that of the COCO
+                official evaluation metric.
+            output_transform: a callable that is used to transform the
+                :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+                form expected by the metric. This can be useful if, for example, you have a multi-output model and
+                you want to compute the metric with respect to one of the outputs.
+                By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+            device: specifies which device updates are accumulated on. Setting the
+                metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+                non-blocking. By default, CPU.
+        """
+        try:
+            from torchvision.ops import box_iou
+
+            self.box_iou = box_iou
+        except ImportError:
+            raise ModuleNotFoundError("This module requires torchvision to be installed.")
+
+        if iou_thresholds is None:
+            self.iou_thresholds = [
+                0.5,
+                0.55,
+                0.6,
+                0.65,
+                0.7,
+                0.75,
+                0.8,
+                0.85,
+                0.8999999999999999,
+                0.95,
+            ]  # torch.linspace(0.5, 0.95, 10).tolist()
+        elif isinstance(iou_thresholds, torch.Tensor):
+            if iou_thresholds.ndim != 1:
+                raise ValueError(
+                    "`iou_thresholds` should be a one-dimensional tensor or a list of floats"
+                    f", given a {iou_thresholds.ndim}-dimensional tensor."
+                )
+            self.iou_thresholds = iou_thresholds.sort().values.tolist()
```

Comment:
Please add a test for this case

Reply:
For which case exactly? We have `test_wrong_input` now.